pkg/disttask/framework/taskexecutor/execute/interface.go

```diff
@@ -111,3 +116,19 @@ func SetFrameworkInfo(exec StepExecutor, step proto.Step, resource *proto.StepRe
 		info.Set(reflect.ValueOf(toInject))
 	}
 }
+
+// ModifyResource modifies the resource of the StepExecutor.
+func ModifyResource(exec StepExecutor, resource *proto.StepResource) {
+	if exec == nil {
+		return
+	}
+	e := reflect.ValueOf(exec)
```

Comment:
We dont' need to use reflection 😂 Can we add a private `setResource` method to `StepExecFrameworkInfo`. And use it by `exec.setResource`?

`SetFrameworkInfo` is different, because the `StepExecFrameworkInfo` is nil and it can only be initialized by `SetFrameworkInfo` using reflection